File: openoversea/src/overseainternal/java/com/bytedance/sdk/open/tiktok/TikTokOpenApiFactory.java
package com.bytedance.sdk.open.tiktok;

import android.app.Activity;
import android.text.TextUtils;

import androidx.annotation.Nullable;

import com.bytedance.sdk.open.tiktok.authorize.AuthService;
import com.bytedance.sdk.open.tiktok.common.handler.IApiEventHandler;
import com.bytedance.sdk.open.tiktok.share.ShareImpl;
import com.bytedance.sdk.open.tiktok.api.TikTokOpenApi;
import com.bytedance.sdk.open.tiktok.impl.TikTokOpenApiImpl;

public class TikTokOpenApiFactory {

    private static TikTokOpenConfig sConfig;

    public static boolean init(TikTokOpenConfig config) {
        if (config != null && !TextUtils.isEmpty(config.clientKey)) {
            sConfig = config;
            return true;
        }
        return false;
    }
    private static TikTokOpenApiImpl sharedApiImpl;
    public static TikTokOpenApi create(Activity activity) {
        return create(activity, null);
    }
    /**
     * 创建 TTOpenApi
     *
     * @param activity
     * @return
     */
    public static TikTokOpenApi create(Activity activity, @Nullable IApiEventHandler handler) {
        if (sConfig == null) {
            return null;
        }
        synchronized (TikTokOpenApiFactory.class) {
            if (sharedApiImpl == null) {
                ShareImpl share = new ShareImpl(activity, sConfig.clientKey);
                AuthService authService = new AuthService(activity, sConfig.clientKey);
                sharedApiImpl = new TikTokOpenApiImpl(activity, authService, share, handler);
            }
            return sharedApiImpl;
        }
    }
}

File: openoversea/src/overseainternal/java/com/bytedance/sdk/open/tiktok/TikTokOpenApiFactory.kt
package com.bytedance.sdk.open.tiktok

/*
    Copyright 2022 TikTok Pte. Ltd.

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
 */

import android.content.Context
import android.text.TextUtils
import com.bytedance.sdk.open.tiktok.api.TikTokOpenApiInternal
import com.bytedance.sdk.open.tiktok.authorize.AuthService
import com.bytedance.sdk.open.tiktok.common.handler.IApiEventHandler
import com.bytedance.sdk.open.tiktok.impl.TikTokOpenApiInternalImpl
import com.bytedance.sdk.open.tiktok.share.ShareService

class TikTokOpenApiFactory {
    companion object {
        private var sConfig: TikTokOpenConfig? = null

        @JvmStatic
        fun init(config: TikTokOpenConfig): Boolean {
            if (!TextUtils.isEmpty(config.clientKey)) {
                sConfig = config
                return true
            }
            return false
        }

        @JvmStatic
        fun create(context: Context, handler: IApiEventHandler): TikTokOpenApiInternal {
            if (sConfig == null) {
                throw Exception("Please init TikTokOpenApiFactory first before creating api")
            }
            checkNotNull(sConfig).let {
                val shareService = ShareService(context, it.clientKey)
                val authService = AuthService(context, it.clientKey)
                return TikTokOpenApiInternalImpl(context, authService, shareService, handler)
            }
        }
    }
}
